### PR TITLE
Add MIT licence headers, remove obsolete files

### PR DESCRIPTION
**Pull request by dankwart-de**

### CODE REVIEW

File: src/de/dkfz/b080/co/common/WorkflowUsingMergedBams.groovy [hunk 1]
```diff
@@ -4,191 +4,171 @@
  * Distributed under the MIT License (license terms are at https://www.github.com/eilslabs/COWorkflowsBasePlugin/LICENSE.txt).
  */
 
-package de.dkfz.b080.co.common
+package de.dkfz.b080.co.common;
```
Comment (vinjana) [inline src/de/dkfz/b080/co/common/WorkflowUsingMergedBams.groovy]: Why did you add all the `;` to a Groovy file?

File: src/de/dkfz/b080/co/common/WorkflowUsingMergedBams.groovy [hunk 2]
```diff
@@ -4,191 +4,171 @@
...
-import static de.dkfz.b080.co.files.COConstants.FLAG_EXTRACT_SAMPLES_FROM_OUTPUT_FILES
+import static de.dkfz.b080.co.files.COConstants.FLAG_EXTRACT_SAMPLES_FROM_OUTPUT_FILES;
 
 /**
  * A basic workflow which uses merged bam files as an input and offers some check routines for those files.
  * Created by michael on 05.05.14.
  */
 @CompileStatic
-abstract class WorkflowUsingMergedBams extends Workflow {
+public abstract class WorkflowUsingMergedBams extends Workflow {
```
Comment (vinjana) [inline src/de/dkfz/b080/co/common/WorkflowUsingMergedBams.groovy]: Again Groovy to Java change for a Groovy file? Is this kind of a reimport?

File: src/de/dkfz/b080/co/common/WorkflowUsingMergedBams.groovy (hunk 1 above)
Comment (dankwart-de) [reply src/de/dkfz/b080/co/common/WorkflowUsingMergedBams.groovy]: Ah that was pretty weird. I don't know why but things got mixed up. This should be the latest version of the code. And then also I had a different version of the code, which was only half done... I'll clean it up and add more comments. And yes, to your question below, it is somehow a reimport.

File: src/de/dkfz/b080/co/common/WorkflowUsingMergedBams.groovy [hunk 3]
```diff
@@ -94,17 +100,37 @@ public abstract class WorkflowUsingMergedBams extends Workflow {
...
-                context.addErrorEntry(ExecutionContextError.EXECUTION_NOINPUTDATA.expand("Control bam is missing and workflow is not set to accept tumor only"));
-            if (bamTumorMerged == null)
-                context.addErrorEntry(ExecutionContextError.EXECUTION_NOINPUTDATA.expand("Tumor bam is missing"));
-            return false;
+        if (!initialBamFiles) {
+            context.addErrorEntry(ExecutionContextError.EXECUTION_NOINPUTDATA.expand("Did not find any bam files."));
+            return false
+        }
+        if (isNoControlWorkflow) {
+            boolean foundAll = true
```
Comment (vinjana) [inline src/de/dkfz/b080/co/common/WorkflowUsingMergedBams.groovy]: Should be `boolean allAreTumor = true` or so.

File: src/de/dkfz/b080/co/common/COConfig.groovy [hunk 4]
```diff
@@ -12,83 +18,83 @@ class COConfig {
...
     }
 
     // This is used so often, it should maybe be part of ExecutionContext.
-    public RecursiveOverridableMapContainerForConfigurationValues getConfigValues() {
+    RecursiveOverridableMapContainerForConfigurationValues getConfigValues() {
         return context.getConfiguration().getConfigurationValues()
     }
 
-    public setConfig(String flagName, String value, String typeName) {
+    def setConfig(String flagName, String value, String typeName) {
```
Comment (vinjana) [inline src/de/dkfz/b080/co/common/COConfig.groovy]: void setConfig() better expresses, that this is a modifying op.

Comment (dankwart-de) [reply src/de/dkfz/b080/co/common/COConfig.groovy]: I'll fix it :D. That's actually nothing I changed by myself. The code formatter did it and formerly void was missing... though it should be there for clearness! Groovy can be fun.
